examples/docker-compose/docker-compose.yaml

```diff
@@ -0,0 +1,18 @@
+---
+version: "3.8"
+
+services:
+  ollama:
+    build:
+      context: ../..
+      dockerfile: Dockerfile 
+    container_name: ollama
+    image: ollama/ollama
+    healthcheck:
+      test: ["CMD-SHELL", "cat < /dev/null > /dev/tcp/localhost/11434 || exit 1"]
```

Comment:
I believe /dev/tcp only works with bash.  Since the master Dockerfile doesn't set SHELL, the default shell is /bin/sh, which doesn't support  this operation.

Reply:
This is an excellent comment. In this case, the base image is Ubuntu:

$ docker run -it --entrypoint /bin/sh ollama/ollama
# cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.3 LTS"

Which means, that the system shell /usr/bin/sh is linked to dash

# ls -l /usr/bin/sh
lrwxrwxrwx 1 root root 4 Mar 23  2022 /usr/bin/sh -> dash

Testing with the system shell (dash) indeed does not support the /dev/tcp mechanism for creating TCP connections:

# cat < /dev/null > /dev/tcp/localhost/11434
/bin/sh: 7: cannot create /dev/tcp/localhost/11434: Directory nonexistent
# echo $?
2

However using bash, we are able to support this health-check:

# bash
root@0026649d68b2:/# cat < /dev/null > /dev/tcp/localhost/11434
root@0026649d68b2:/# echo $?
0

Thus, we should be specifying bash explicitly in the compose yml file, with the health-check. 

SHELL ["/bin/bash", "-c"]

Which I'll add to the configuration. 

Thanks again for this insightful comment!

Reply:
Just a further comment after additional investigation, indeed the health-check fails using the original configuration:

$ docker ps
CONTAINER ID   IMAGE           COMMAND               CREATED              STATUS                          PORTS                                           NAMES
075df43c3f46   ollama/ollama   "/bin/ollama serve"   About a minute ago   Up About a minute (unhealthy)   0.0.0.0:11434->11434/tcp, :::11434->11434/tcp   ollama

Notice (unhealth). 

Changing the health-check to explicitly use bash in the compose file, returns a healthy status:

      test: ["CMD-SHELL", "/usr/bin/bash -c 'cat < /dev/null > /dev/tcp/localhost/${OLLAMA_PORT:-11434} || exit 1'"]
      
Returns a successful (healthy) status:

$ docker ps
CONTAINER ID   IMAGE           COMMAND               CREATED         STATUS                   PORTS                                           NAMES
7b0498b23ea4   ollama/ollama   "/bin/ollama serve"   2 minutes ago   Up 2 minutes (healthy)   0.0.0.0:11434->11434/tcp, :::11434->11434/tcp   ollama

So, again - great call out.